src/app/sponsors/sponsors.component.ts

```diff
@@ -39,7 +49,22 @@ export class SponsorsComponent implements OnInit {
   }
 
   ngOnInit() {
+    // Initialize sponsors and partners
     this.sponsors = this.sponsorsService.getSponsors();
+    this.partners = this.sponsorsService.getPartners();
+
+    // Hide second row if necessary
+    this.hideSecondRow();
+  }
+
+  hideSecondRow() {
```

Comment:
check how many times this is executed

Reply:
Ya I was calling it once during the oninit, when I only really need to call it once on route change. This isn't too big of a deal since it is only called one extra time on full refresh of a page, but I have removed it since it is redundant.